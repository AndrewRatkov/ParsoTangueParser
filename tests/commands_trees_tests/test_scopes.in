int x := 12 + 18;
int y := x + x;
if x < y then
str z := "Hello, World!";
else
int z := y - x;
fi