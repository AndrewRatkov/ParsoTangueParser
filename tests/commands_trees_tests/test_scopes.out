[DEF int x]--->[INT,+ ]------->[INT,12]
                   |
                   +---------->[INT,18]
[DEF int y]--->[INT,+ ]------->[INT,x]
                   |
                   +---------->[INT,x]
IF
|
+--->[INT,< ]------->[INT,x]
|        |
|        +---------->[INT,y]
|
THEN
|
+--->[DEF str z]--->[STR,"Hello, World!"]
|
ELSE
|
+--->[DEF int z]--->[INT,- ]------->[INT,y]
|                       |
|                       +---------->[INT,x]
|
FI
